src/util/tprint.cpp

```diff
@@ -47,6 +47,7 @@ u32 smek_snprint(char *out_buffer, u32 buf_size, const char *in_buffer) {
 
 template<>
 i32 sntprint<>(char *buffer, u32 buf_size, const char *fmt) {
+    if (buf_size == 0) return 0;
```

Comment:
Even though this is tempting, we now get a string that isn't null terminated. This we cannot allow since it can cause all manner of serious bugs. We should crash on this instead, since it doesn't make sense to write to a 0-length string except when formatting, and then we can write to negative values.

I see 2 fixes:
- Assert that the buffer size is never 0.
- Ignore that the buffer size is 0 and write a null terminator.

Reply:
Currently (on `master`), we have [an equivalent check to the proposed in `sntprint_helper`](https://github.com/FredTheDino/SMEK/blob/6d179efb7ccc6778794a899ed3ecfed953efa745/src/util/tprint.h#L79).

1) Asserting `buf_size != 0` trips the following call, which I don't think it should.
```cpp
char buffer[5];
sntprint(buffer, 5, "123{}", 45);
```
We can't really fail earlier since `sntprint` is (by design) reachable as is.

2) Is this what you mean?
```cpp
if (buf_size == 0) {
    buffer[0] = '\0';
    return 0;
}
```
If so, I guess it works but now we're ignoring what `buf_size` "means", i.e. "the maximum number of characters that can safely be written to the buffer" (if we adopt the same meaning as `std::snprintf`). Arguably a nonissue, but if I tell a function "you may not write anything to this pointer" I'd feel a bit stumped if it segfaults because it tries to write `\0` anyway.

Reply:
My proposed fix more along the lines of:
```
if (buf_size == 0) {
    buffer[-1] = '\0';
    return 0;
}
```
So your given case of:
```
char buffer[5];
sntprint(buffer, 5, "123{}", 45);
```
Would produce the output: "1234\0", instead of "12345" without the null terminator. Since we interface a lot with C-functions missing the nullterminator might cause issues further down. But this isn't a perfect solution.

I still think the assert is better, since it trips the edge case where we don't write the null terminator. All i'm saying, is we need the null terminator.